server/domain/services/UserService.ts

```diff
@@ -1,22 +1,55 @@
-import { UserDto, UserUpdateDto } from '@domain/DTO';
+import { UserCreateDto, UserDto, UserUpdateDto } from '@domain/DTO';
 import IUsersRepository from '@domain/interfaces/repositories/IUserRepository';
 import IUsersService from '@domain/interfaces/services/IUserService';
-import { EntityNotFoundError } from '@domain/models';
+import { CustomError, EntityNotFoundError, User } from '@domain/models';
 import { injectable } from 'inversify';
 
 @injectable()
 export default class UserService implements IUsersService {
-  constructor(private UserRepository: IUsersRepository) {}
+  constructor(private repository: IUsersRepository) {}
 
   async search(): Promise<UserDto[]> {
-    return this.UserRepository.search();
+    return this.repository.search();
   }
 
-  async update(id: number, body: UserUpdateDto): Promise<UserDto> {
-    const userExists = await this.UserRepository.getById(id);
+  async create(dto: UserCreateDto): Promise<User> {
+    const emailAlreadyExists = await this.repository.verifyIfEmailAlreadyExists(dto.email);
 
-    if (!userExists) throw new EntityNotFoundError('user', id);
+    if (emailAlreadyExists) {
+      throw new CustomError('Email already exists', 404);
```

Comment:
`throw new CustomError('Email already exists', 404);`
change to
`throw new CustomError('Email already exists', 401);`